src/utils/extension.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface IDockerExtensionPackageJson {
+    name: string;
+    version: string;
+    publisher: string;
+}
+
+let packageJson: IDockerExtensionPackageJson;
+export function getDockerExtensionPackageJson(): IDockerExtensionPackageJson {
+    if (!packageJson) {
+        // Don't remove the following constant, or webpack will complain about nonexistent package.
+        const packageJsonPath = '../../../package.json';
+        packageJson = require(packageJsonPath) as IDockerExtensionPackageJson;
+    }
```

Comment:
`vscode.extensions.getExtension('<id>')` should provide access to the same information (canonical name and package JSON); not sure if that's any simpler.  We already embed our extension ID as a constant for that purpose.

Reply:
I missed that--will use the hard-coded extension ID.